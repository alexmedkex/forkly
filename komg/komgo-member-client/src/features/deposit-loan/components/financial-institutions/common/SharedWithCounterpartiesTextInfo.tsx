import * as React from 'react'
import { IDepositLoanForm } from '../../../store/types'

interface IProps {
  requested?: boolean
  values?: IDepositLoanForm
}

const SharedWithCounterpartiesTextInfo: React.FC<IProps> = (props: IProps) => {
  if (props.requested) {
    return (
      <React.Fragment>
        <h3>Requests to disclose information</h3>
        <p>
          The counterparties below requested information on {props.values ? props.values.currencyAndTenor : ''} to be
          shared with them. Choose what data to disclose to individual counterparts. This data will be disclosed
          bilaterally as non-binding and does not represent or warrant any commitment on your part.
        </p>
      </React.Fragment>
    )
  }
  return (
    <React.Fragment>
      <h3>Visibility to counterparties</h3>
      <p>
        Information shared with your counterparties on a bilateral basis. This information is displayed to
        counterparties as non-binding and does not represent or warrant any commitment on your part.
      </p>
    </React.Fragment>
  )
}

export default SharedWithCounterpartiesTextInfo
