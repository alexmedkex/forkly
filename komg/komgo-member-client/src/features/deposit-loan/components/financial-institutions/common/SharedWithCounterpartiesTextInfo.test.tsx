import * as React from 'react'
import * as renderer from 'react-test-renderer'

import SharedWithCounterpartiesTextInfo from './SharedWithCounterpartiesTextInfo'

describe('SharedWithCounterpartiesTextInfo', () => {
  it('should match default snaptshout', () => {
    expect(renderer.create(<SharedWithCounterpartiesTextInfo />).toJSON()).toMatchSnapshot()
  })
})
