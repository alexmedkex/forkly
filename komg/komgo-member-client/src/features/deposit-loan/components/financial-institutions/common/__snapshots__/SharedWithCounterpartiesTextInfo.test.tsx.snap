// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SharedWithCounterpartiesTextInfo should match default snaptshout 1`] = `
Array [
  <h3>
    Visibility to counterparties
  </h3>,
  <p>
    Information shared with your counterparties on a bilateral basis. This information is displayed to counterparties as non-binding and does not represent or warrant any commitment on your part.
  </p>,
]
`;
